src/server/_common.py

```diff
@@ -125,7 +125,9 @@ def before_request_execute():
         real_remote_addr=get_real_ip_addr(request),
         user_agent=request.user_agent.string,
         api_key=api_key,
-        user_id=(user and user.id)
+        user_id=(user and user.id),
+        req_referrer=request.referrer,
+        req_origin=request.environ.get('HTTP_ORIGIN', '')
```

Comment:
- Can you move this to right underneath `user_agent=...` (since they come from the same place (request headers) and because they both identify client context)?

- Why not just use [`request.origin`](https://flask.palletsprojects.com/en/3.0.x/api/#flask.Request.origin)?  Presumably so you can default to the empty string, but why prefer that over `None`?

- We dont need to keep both values; if both are provided, refer[r]er should be a superstring of origin.
```suggestion
        refer_origin=request.referrer or request.origin,
```

Reply:
Fixed!